File: python/point_time_units.py
from requests import post
from numpy import float64
from numpy.ma import masked_array
from os import getenv
from datetime import datetime


# NOTE: don't for get to set "apikey" env, or the default below.
resp = post(
    "https://forecast-v2.metoceanapi.com/point/time",
    headers={"x-api-key": getenv("apikey", "MYAPIKEY")},
    json={
        "points": [{
            "lon": 174.7842,
            "lat": -37.7935
        }],
        "variables": [
            "air.temperature.at-2m"
        ],
        "time": {
            "from": "{:%Y-%m-%dT00:00:00Z}".format(datetime.utcnow()),
            "interval": "3h",
            "repeat": 2
        }
    }
)

if resp.status_code != 200:
    raise ValueError("{}: {}", resp.status_code, resp.text)

data = resp.json()
var = data["variables"]["air.temperature.at-2m"]

# You will usually want to make a masked arrays with the "noData" field. 0 == Good data, any other value indicates
# the "data" value is null, look up the field "noDataReasons" to find out why.
temperature = masked_array(var["data"], mask=var["noData"], dtype=float64)
units = var["units"]

# Convert Kelvin to Degree C.
# You have a couple of options for converting units. https://forecast-v2.metoceanapi.com/units/
# You can hand code it seen below.
# Or you can look up the units from the Forecast-API and look up the "conversions" field, Note: always apply multiplier
# before offset.
# degreeK:
#     ...
#     conversions:
#       degreeC:
#         offset: -273.15
#       degreeF:
#         offset: -459.67
#         multiplier: 1.8

print("{}: {}".format(units, temperature))

if units == "degreeK":
    temperature += -273.15

elif units == "degreeF":
    temperature *= 0.555556
    temperature += -17.777778

print("degreeC: {}".format(temperature))

File: point-time-units.js
let url = 'https://forecast-v2.metoceanapi.com/point/time'

let data = {
  points: [{lon: 174.7842, lat: -37.7935}],
  variables: ['air.temperature.at-2m'],
  time: {
    from: new Date().toISOString(),
    interval: '3h',
    repeat: 3,
  }
}

function multiplierOffsetToDegreesC(srcUnits) {
    switch (srcUnits) {
    case 'degreeK':
      return {multiplier: 1, offset: -273.15}
    case 'degreeC':
      return {multiplier: 1, offset: 0}
    case 'degreeF':
      return {multiplier: 0.555556, offset: -17.777778}
    }
  return {error: 'Can\'t convert data to degreesC from ' + srcUnits}
}

/*
This demonstrates how to detect the units a variable's data are in,
and one way to convert them.
Some conversion information can also be retrieved from the
https://forecast-v2.metoceanapi.com/units/ endpoint, which returns JSON
with the same multiplier/offset format used here
*/
export let pointTimeUnits = {
  url,
  data,
  cb: function(data) {
    let temp = data['variables']['air.temperature.at-2m']
    const transform = multiplierOffsetToDegreesC(temp['units'])
    if (transform.error) {
      return transform.error
    }

    return {'air.temperature.at-2m': {
      data: temp['data'].map((datum, index) => {
        if (temp['noData'][index]) {
          return null
        }
        return transform.multiplier * datum + transform.offset
      }),
      'noData': temp['noData'],
      'units': 'degreeC',
    }}
  }
}
